community/kernel/src/main/java/org/neo4j/helpers/Args.java

```diff
@@ -249,29 +249,57 @@ public long getDuration( String key, long defaultValueInMillis)
         return value != null ? TimeUtil.parseTimeMillis.apply(value) : defaultValueInMillis;
     }
 
-    public Boolean getBoolean( String key, Boolean defaultValue )
+    /**
+     * Like calling {@link #getBoolean(String, Boolean)} with {@code false} for default value.
+     * This is the most common case, i.e. returns {@code true} if boolean argument was specified as:
+     * <ul>
+     * <li>--myboolean</li>
+     * <li>--myboolean=true</li>
+     * </ul>
+     * Otherwise {@code false.
+     * }
+     * @param key argument key.
+     * @return {@code true} if argument was specified w/o value, or w/ value {@code true}, otherwise {@code false}.
+     */
+    public boolean getBoolean( String key )
     {
-        String value = getSingleOptionOrNull( key );
+        return getBoolean( key, false );
+    }
 
-        // Apparently this condition must be split like this, instead of as an elvis operator,
-        // because defaultValue will, in that case, be evaluated as a primitive boolean and
-        // a NullPointerException will insue. Odd.
-        if ( value != null )
-        {
-            return Boolean.parseBoolean( value );
-        }
-        return defaultValue;
+    /**
+     * Like calling {@link #getBoolean(String, Boolean, Boolean)} with {@code true} for
+     * {@code defaultValueIfNoValue}, i.e. specifying {@code --myarg} will interpret it as {@code true}.
+     */
+    public Boolean getBoolean( String key, Boolean defaultValueIfNotSpecified )
+    {
+        return getBoolean( key, defaultValueIfNotSpecified, Boolean.TRUE );
```

Comment:
I'm wondering if the `defaultValueIfSpecifiedButNoValue` should be `! defaultValueIfNotSpecified`, to enable parameters such as:

`myApp --disableSomething`

to be used as:

`args.getBoolean( "disableSomething", true )`